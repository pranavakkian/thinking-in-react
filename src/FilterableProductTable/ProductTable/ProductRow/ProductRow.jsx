import PropTypes from 'prop-types';
import React, { memo } from 'react';

const ProductRow = (props) => {
  const { data } = props;
  const { name, price } = data;

  return (
    <>
      <td id="dataName">{name}</td>
      <td id="dataPrice">{price}</td>
    </>
  );
};

ProductRow.propTypes = {
  data: PropTypes.shape({
    category: PropTypes.string.isRequired,
    price: PropTypes.string.isRequired,
    stocked: PropTypes.bool.isRequired,
    name: PropTypes.string.isRequired,
  }).isRequired,
};

export default memo(ProductRow);
