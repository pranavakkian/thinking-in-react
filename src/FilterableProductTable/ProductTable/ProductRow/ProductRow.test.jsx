/* eslint-disable no-undef */
import { mount } from 'enzyme';
import React from 'react';
import ProductRow from './ProductRow';

const data = {
  category: 'Food',
  price: '$10',
  stocked: true,
  name: 'Apple',
};

const wrapper = mount(<ProductRow data={data} />);

describe('ProductRow', () => {
  it('ProductRow rendered successfully!', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Product row data rendered successfully!', () => {
    const tableData = wrapper.find('td');
    expect(tableData).toHaveLength(2);
    const dataName = wrapper.find('#dataName');
    expect(dataName.text()).toEqual('Apple');
    const dataPrice = wrapper.find('#dataPrice');
    expect(dataPrice.text()).toEqual('$10');
  });

  it('ProductRow un-mounted successfully!', () => {
    wrapper.unmount();
  });
});
