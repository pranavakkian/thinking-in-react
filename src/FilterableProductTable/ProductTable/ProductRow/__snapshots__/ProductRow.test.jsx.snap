// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductRow ProductRow rendered successfully! 1`] = `
<Memo(ProductRow)
  data={
    Object {
      "category": "Food",
      "name": "Apple",
      "price": "$10",
      "stocked": true,
    }
  }
>
  <td
    id="dataName"
  >
    Apple
  </td>
  <td
    id="dataPrice"
  >
    $10
  </td>
</Memo(ProductRow)>
`;
